c2corg_api/views/user.py

```diff
@@ -67,15 +67,18 @@ def validate_json_password(request):
         request.errors.add('body', 'password', 'Invalid')
 
 
+def available_user_attribute(attrname, value):
+    attr = getattr(User, attrname)
+    return DBSession.query(User).filter(attr == value).count() == 0
+
+
 def validate_unique_attribute(attrname, request):
     """Checks if the given attribute is unique.
     """
 
     if attrname in request.json:
         value = request.json[attrname]
-        attr = getattr(User, attrname)
-        count = DBSession.query(User).filter(attr == value).count()
-        if count == 0:
+        if available_user_attribute(attrname, value):
```

Comment:
When calling the function we are actually intersted in knowing if this value is available (unused yet).
I considered a lower level name like `would_violate_unique_constraint_for_user_attribute` though I felt it a bit too long.
What about `is_available_user_attribute` or `is_unused_user_attribute`?